### DISCUSSION TITLE
When dispatching event on livewire component with form, event is dispatched after validation (even if put before validation)

### DISCUSSION THREAD

**Discussion by BoGnY** [livewire/livewire#8598] [7. Bugs (archived)] [answered]
### Livewire version

v3.5.1

### Laravel version

v10.48.4

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. dispatch event
2. call form validation
3. dispatch another event

the first event is dispatched after validation together the second event instead dispatching when called (before validation)

### Repository or Wirebox URL

https://wirebox.app/b/47mkn

### Additional information

Minimal explaination here usernotnull/tall-toasts#546 (toast dispatch an event)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by samlev** [comment, marked as answer]
This isn't a bug - it's just how the lifecycle works. Calling `$this->dispatch()` just adds to the list of events that will be dispatched _after the current request completes_.

If you want "real time" events that will update the frontend _before_ the livewire request completes, you'll need to use websockets.

**Reply by BoGnY** [reply]
Thanks 👍🏻
